torch/lib/THCS/generic/THCSTensorMath.cu

```diff
@@ -460,6 +462,115 @@ void THCSTensor_(cmul)(THCState *state, THCSTensor *r_, THCSTensor *t_, THCSTens
   THCSTensor_(free)(state, src);
 }
 
+
+ void THCTensor_(spbqrfactsolve)(THCState *state, THCTensor *rx_, THCTensor *b, THCIndexTensor *Ai, THCTensor *Av, THLongStorage *Asz) {
+#if defined(THCS_REAL_IS_FLOAT) || defined(THCS_REAL_IS_DOUBLE)
+   THCAssertSameGPU(THCSTensor_(checkGPU)(state, 1, 3, rx_, b, Av));
+
+   THArgCheck(b->nDimension == 2, 2,
+              "Batch tensor b expected, got %dD tensor", b->nDimension);
+   THArgCheck(Av->nDimension == 2, 2,
+              "Batch tensor Av expected, got %dD tensor", Av->nDimension);
+
+   long nBatch = THCTensor_(size)(state, Av, 0);
+   long nnz = THCTensor_(size)(state, Av, 1);
+   long m = Asz->data[0];
+   long n = Asz->data[1];
+
+   THCTensor_(resize2d)(state, rx_, nBatch, m);
+
+   THCIndexTensor *rowIndices = THCIndexTensor_(newSelect)(state, Ai, 0, 0);
+   THCIndexTensor *colIndices = THCIndexTensor_(newSelect)(state, Ai, 0, 1);
+   THCudaIntTensor *csr = THCSTensor_(toCSR)(state, rowIndices, m, nnz);
+   THCudaIntTensor *colIndicesInt = THCudaIntTensor_newWithSize1d(state, colIndices->size[0]);
+   THCudaIntTensor_copyCudaLong(state, colIndicesInt, colIndices);
```

Comment:
This should probably be preceded by a `max` operation that checks that this conversion won't overflow. It might cause some weird silent errors otherwise